src/nodeClient.js

```diff
@@ -1,93 +1,64 @@
-const Web3 = require('web3');
+const ethers = require('ethers');
 const { helpers } = require('leap-core');
 
 const { formatHostname, advanceBlocks, sleep } = require('./helpers');
 
 let idCounter = 0;
 
-class Node {
+class Node extends helpers.LeapEthers {
   constructor(hostname, jsonrpcPort) {
-    this.id = idCounter++;
-    this.web3 = helpers.extendWeb3(new Web3(formatHostname(hostname, jsonrpcPort)));
-  }
+    const provider = new ethers.providers.JsonRpcProvider(formatHostname(hostname, jsonrpcPort))
+    super(provider);
 
-  async send(method, params) {
-    return await new Promise(
-      (resolve, reject) => {
-        this.web3.currentProvider.send(
-          { jsonrpc: '2.0', id: 42, method: method, 'params': params },
-          (err, res) => { if (err) { return reject(err); } resolve(res); }
-        );
-      }
-    );
+    this.id = idCounter++;
   }
 
   async sendTx(tx) {
-    // workaround different transaction hashes (leap-core bug)
-    let txHash;
-
-    try {
-      // web3 hangs here on invalid txs, trying to get receipt?
-      // await this.web3.eth.sendSignedTransaction(tx.hex());
-      const resp = await new Promise(
-        (resolve, reject) => {
-          this.web3.currentProvider.send(
-            { jsonrpc: '2.0', id: 42, method: 'eth_sendRawTransaction', 'params': [tx.hex()] },
-            (err, res) => { if (err) { return reject(err); } resolve(res); }
-          );
-        }
-      );
-      txHash = resp.result;
-    } catch(e) {
-      console.log(e);
-    }
-
-    if (txHash !== tx.hash()) {
-      console.warn(`txHash(${txHash}) from rpc does not match local tx.hash(${tx.hash()})`);
-    }
+    return helpers.sendSignedTransaction(this.provider, tx.hex());
+  };
 
-    let rounds = 50;
-    while (rounds--) {
-      let res = await this.web3.eth.getTransaction(txHash)
+  async getBalance(addr) {
```

Comment:
For the future: `getBalance`, `getBlock` and `getTransaction` could go straight to LeapEthers :)

Reply:
LeapEthers doesn't have those methods implemented yet 🤔

Reply:
yeah.. I think it was meant for leap-specific things. But if these methods are useful here, chances are they might be useful elsewhere

Reply:
Sorry, understood it wrong. Will include them into the leap-core PR 🤦‍♂

Reply:
@troggy 
```
 return this.provider.getBalance(addr).then(res => Number(res));
```
How would we deal with that in LeapEthers? We probably should leave it a BigInt? :)

Reply:
Not sure. I guess casing to Number is wrong — no proper arithmetic will be possible for large values, no?

Also I think if it is just `return this.provider.getBalance(addr)` then we don't need it in LeapEthers :)

Reply:
Ok. Let's leave these in integration tests for now.
Casting to unsafe numbers is a integration-tests thing anyway 🤣